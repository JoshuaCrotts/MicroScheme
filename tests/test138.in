;
; Jesus, we're going λ crazy here!
;

(define (baz x proc y) ((λ (r s) (- (proc r s))) x y))
(define (bar x y) (λ (r s) (* (+ r (baz 150 (λ (z w) (* z (modulo w z) 300)) s) (+ x y)))))
(define foo (λ (x proc y) ((bar 10 20) (proc x y) 30)))
(foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800))

(* (foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800)) (foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800)))
(+ (foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800)) (foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800)))
(- (foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800)) (foo 50 (λ (x y) (- x y)) (foo 60 (λ (x y) (+ (* 2 x) (/ y 2))) -800)))

