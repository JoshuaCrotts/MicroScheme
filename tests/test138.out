330
108900
660
0
